### PR TITLE
docs(cstorvolume): add cStor volume resource organisation document.

### PR DESCRIPTION
**Pull request by sonasingh46**
Signed-off-by: Ashutosh Kumar <ashutosh.kumar@mayadata.io>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ Table of contents:
 - [Quickly deploy it on K8s and get started](docs/quick.md)
 - [Pool Operations Tutorial](docs/tutorial/intro.md)
 - [High level overview](#overview)
-- [CStor Resource Organisation](docs/developer-guide/cstor-pool-docs.md)
+- [CStor Resource Organisation](docs/developer-guide/cstor-pool.md)
```
Comment (prateekpandey14) [inline README.md:36]: we can add the direct links here for better visibilty  
- [CStor Pool Resource Organisation](docs/developer-guide/cstor-pool.md)
- [CStor Volume Resource Organisation](docs/developer-guide/cstor-volume.md)

Comment (sonasingh46) [reply README.md:36]: One has to go through first the cStor pool always and then the cStor volume. 
Also -- putting more links in the main page can make it look crowded.

File: docs/developer-guide/cstor-volume.md [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+|                    |                 X          X                 |                  |
+|                    |                 X          X                 +------------------+
+|                    |                 XXXXXXXXXXXX
++--------------------+
+
+```
+
+4. Now the CVC-Operator creates following resources as part of CVC reconciliation:
+- A `CStorVolume` (CV) (custom resource).  
+- `k` number of `CStorVolumeReplica` (CVR) (custom resource) where `k` is the replica count specified in the storage class.
```
Comment (prateekpandey14) [inline docs/developer-guide/cstor-volume.md:121]: nit : specified in the StorageClass parameters

File: docs/developer-guide/cstor-volume.md [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+- A deployment known as `cStor target`.
+- A service. 
+
+5. `CStorVolume` replica is reconciled by the corresponding `cStor target` that got created.
+
+6. The created `CStorVolumeReplicas` are placed on `CStorPoolInstances`. Now two `CStorVolumeReplicas` for the given PVC
+will be placed on the same `CStorPoolInstance`(CSPI). (The CVR is labelled with a CSPI UUID -- ths is what placement
+means)
+
```
Comment (prateekpandey14) [inline docs/developer-guide/cstor-volume.md:130]: nit: No two `CStorVolumeReplicas` for the given PVC
will be placed on the same `CStorPoolInstance`(CSPI). (The CVR is labelled with a CSPI UUID -- which means replicas will be distributed across the pools)

File: docs/developer-guide/cstor-volume.md [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+      X                    X                 X
+                           X                 X
+                           XXXXXXXXXXXXXXXXXXX
+
+```
+2. As part of PVC create, the csi-provisioner(of Controller Plugin pod) executes a gRPC method that is implemented by 
+`cstor-csi-plugin` (of controller plugin pod). This method is `CreateVolume`.
+
+3. The `cstor-csi-plugin` (as part of CreateVolume) creates a CStorVolumeConfig(CVC) custom resource that is reconciled 
+by CVC-Operator. It should be noted that CVC-Operator is deployed as part of installing cStor.
```
Comment (prateekpandey14) [inline docs/developer-guide/cstor-volume.md:97]: nit: is deployed as part of cstor-operators installation

File: docs/developer-guide/cstor-volume.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+Following are the list of native Kubernetes resources that enables the cStor CSI volume features:
+- Service (Known as target service)
+- Deployment (Known as cStor Target)
+
+# cStor CSI Driver Overview
+
+OpenEBS cStor CSI driver(or plugin) follows a cetralized plugin architecture. In centralized plugin architeture, two 
+componenets are deployed in following manner:
+1. **Controller Plugin:** It is deployed as a deployment controller and scheduled to run on the master node of
```
Comment (mittachaitu) [inline docs/developer-guide/cstor-volume.md:31]: Controller plugin can run anywhere not only in master.

File: docs/developer-guide/cstor-volume.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+```
+
+4. Now the CVC-Operator creates following resources as part of CVC reconciliation:
+- A `CStorVolume` (CV) (custom resource).  
+- `k` number of `CStorVolumeReplica` (CVR) (custom resource) where `k` is the replica count specified in the 
+StorageClass parameters.
+- A deployment known as `cStor target`.
+- A service. 
+
+5. `CStorVolume` replica is reconciled by the corresponding `cStor target` that got created.
```
Comment (mittachaitu) [inline docs/developer-guide/cstor-volume.md:126]:
```suggestion
5. `CStorVolume` resource is reconciled by the corresponding `cStor target` that got created.
```

File: docs/developer-guide/cstor-volume.md (hunk 5 above)
Comment (prateekpandey14) [reply docs/developer-guide/cstor-volume.md:31]: It is deployed as a sidecar container in statefulset with other CSI specific containers

**Pull request merged**